ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -282,19 +283,68 @@ object BigDiffy extends Command {
-    }
-    else {
-      bigDiffy.keyStats.saveAsTextFile(s"$output/keys")
-      bigDiffy.fieldStats.saveAsTextFile(s"$output/fields")
-      bigDiffy.globalStats.saveAsTextFile(s"$output/global")
+  @BigQueryType.toTable
+  case class KeyStatsBigQuery(key: String, diffType: String, delta: Option[DeltaBigQuery])
+  case class DeltaBigQuery(field: String, left: String, right: String, delta: DeltaValueBigQuery)
+  case class DeltaValueBigQuery(deltaType: String, deltaValue: Option[Double])
+  @BigQueryType.toTable
+  case class GlobalStatsBigQuery(numTotal: Long, numSame: Long, numDiff: Long,
+                                 numMissingLhs: Long, numMissingRhs: Long)
+  @BigQueryType.toTable
+  case class FieldStatsBigQuery(field: String,
+                        count: Long,
+                        fraction: Double,
+                        deltaStats: Option[DeltaStatsBigQuery])
+  case class DeltaStatsBigQuery(deltaType: String,
+                        min: Double, max: Double, count: Long,
+                        mean: Double, variance: Double, stddev: Double,
+                        skewness: Double, kurtosis: Double)
+
+  /** saves stats to either GCS as text, or BigQuery */
+  def saveStats[T](bigDiffy: BigDiffy[T], output: String, withHeader: Boolean = false,
+                   saveToBq: Boolean = false): Unit = {
+    val keyStatsPath = s"$output/keys"
+    val fieldStatsPath = s"$output/fields"
+    val globalStatsPath = s"$output/global"
+    if (!saveToBq) {
+      // saving to GCS, either with or without header
+      if (withHeader) {
+        bigDiffy.keyStats.map(_.toString).saveAsTextFileWithHeader(keyStatsPath, "key\tdifftype")
+        bigDiffy.fieldStats.map(_.toString).saveAsTextFileWithHeader(fieldStatsPath,
+          "field\tcount\tfraction\tdeltaType\tmin" +
+            "\tmax\tcount\tmean\tvariance\tstddev\tskewness\tkurtosis")
+        bigDiffy.globalStats.map(_.toString).saveAsTextFileWithHeader(globalStatsPath,
+          "numTotal\tnumSame\tnumDiff\tnumMissingLhs\tnumMissingRhs")
+      }
+      else {
+        bigDiffy.keyStats.saveAsTextFile(keyStatsPath)
+        bigDiffy.fieldStats.saveAsTextFile(fieldStatsPath)
+        bigDiffy.globalStats.saveAsTextFile(globalStatsPath)
+      }
+    } else {
+      // saving to BQ, header irrelevant
+
+      bigDiffy.keyStats.map(stat =>
+        KeyStatsBigQuery(stat.key, stat.diffType.toString, stat.delta.map(d => {
+          val dv = if(d.delta.isInstanceOf[TypedDelta]) {
+            val typedDelta = d.delta.asInstanceOf[TypedDelta]
+            DeltaValueBigQuery(typedDelta.deltaType.toString, Option(typedDelta.value))
+          } else {
+            DeltaValueBigQuery("UNKNOWN", None)
+          }
+          DeltaBigQuery(d.field, d.left.toString, d.right.toString, dv)})
+        ))
+        .saveAsTypedBigQuery(s"${output}_keys")
+      bigDiffy.fieldStats.map(stat =>
+        FieldStatsBigQuery(stat.field, stat.count, stat.fraction, stat.deltaStats.map(ds =>
```

Comment:
Maybe this transformation could live in a companion to FieldStatsBigquery,  or similar approach?